src/contracts/structs.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{
+    contracts::{Pool, PoolKey, Tick},
+    math::types::{sqrt_price::SqrtPrice, token_amount::TokenAmount},
+};
+use ink::prelude::vec::Vec;
+
+#[derive(Default, Clone, Debug, PartialEq)]
+#[ink::scale_derive(Encode, Decode, TypeInfo)]
+pub struct CalculateSwapResult {
```

Comment:
I think the file name could be renamed. It only contains structs related to swap.